processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1010,6 +1024,17 @@ public QueryableIndex load(File inDir) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
+      Map<String, Object> metadata = null;
+      ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
+      if (metadataBB != null) {
+        metadata = mapper.readValue(
+            serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
```

Comment:
It looks like they definitely will cause a hard failure; are you asking if they *should*? I dunno, I think it could go either way but I lean towards probably not. The justification for that being that if anything out there really cares deeply about metadata, then that thing show blow up if it's null, rather than the index mapping itself blowing up.